src/main/scala/clocks/ClockDomain.scala

```diff
@@ -0,0 +1,31 @@
+package freechips.rocketchip.clocks
+
+import chisel3._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+
+class ClockSinkDomain(take: Option[ClockParameters] = None)(implicit p: Parameters) extends LazyModule
+    with LazyScope
+    with HasClockDomainCrossing {
+
+  val node = ClockSinkNode(Seq(ClockSinkParameters(take = take)))
+
+  lazy val module = new LazyRawModuleImp(this) {
+    val (bundle, edge) = node.in.head
+    childClock := bundle.clock
+    childReset := bundle.reset
+  }
+}
+
+class ClockSourceDomain(give: Option[ClockParameters] = None)(implicit p: Parameters) extends LazyModule
+    with LazyScope
+    with HasClockDomainCrossing {
+
+  val node = ClockSourceNode(Seq(ClockSourceParameters(give = give)))
+
+  lazy val module = new LazyRawModuleImp(this) {
+    val (bundle, edge) = node.out.head
+    childClock := bundle.clock
+    childReset := bundle.reset
+  }
+}
```

Comment:
what's the use case for these two classes? is the idea you'd use `LazyScope` to put modules sharing a clock into the same `ClockSourceDomain` hierarchy, for example?

Reply:
Yes, you can wrap arbitrary subgraphs of adapters with it and give them all the same clock, similar to what the `couple` methods do in the `TLBusWrapper`s but with added clock assignments.